src/Microsoft.Health.Dicom.Core/Features/Model/VersionedInstanceIdentifier.cs

```diff
@@ -18,6 +18,21 @@ public class VersionedInstanceIdentifier : InstanceIdentifier
         Version = version;
     }
 
+    public VersionedInstanceIdentifier(
+        string studyInstanceUid,
+        string seriesInstanceUid,
+        string sopInstanceUid,
+        string modalitiesInStudy,
+        long version,
+        int partitionKey = default)
+        : base(studyInstanceUid, seriesInstanceUid, sopInstanceUid, partitionKey)
+    {
+        Version = version;
+        ModalitiesInStudy = modalitiesInStudy;
```

Comment:
This needs to go to a separate response model.  I created a draft to show how. I will create more skeleton in January if it helps

Reply:
Thank you, that will be very helpful! I specifically created this draft PR of our PoC in order to collect these feedbacks before any actually commit. I myself will also be on vacation shortly, so looking forward to your skeleton in January.